cocotb/share/makefiles/simulators/Makefile.icarus

```diff
@@ -54,11 +54,20 @@ endif
 
 BUILD_VPI=1
 
-COMPILE_ARGS += -g2012 # Default to latest SystemVerilog standard
+# Do not directly append to COMPILE_ARGS, SIM_ARGS, EXTRA_ARGS etc.,
+# because if a user sets these variables on the make command line,
+# only the user's variable values will be seen.
+# Instead, append to local variables.
+_COMPILE_ARGS += $(COMPILE_ARGS)
+_SIM_ARGS     += $(SIM_ARGS)
+_EXTRA_ARGS   += $(EXTRA_ARGS)
+_PLUSARGS     += $(PLUSARGS)
+
+_COMPILE_ARGS += -g2012 # Default to latest SystemVerilog standard
```

Comment:
This is exactly the type of case where we _do_ want the user to be able to override, as otherwise they have no way to replace this.

Reply:
I'd say that *this* option should not be in this simulator Makefile at all, like all other non-general assumptions.
But I now realized that the approach of this PR does not hold up well at all in the case where you would try to set this option in the Makefile of the testcase, which you'd do with ``COMPILE_ARGS``, and if you then try to append to it via ``make COMPILE_ARGS=-foo``, you are back to square one. So indeed the only solution is to educate users about setting this as an environment variable.